src/bytes_mut.rs

```diff
@@ -1549,6 +1549,23 @@ fn vptr(ptr: *mut u8) -> NonNull<u8> {
     }
 }
 
+/// Returns a dangling pointer with the given address. This is used to store
+/// integer data in pointer fields.
+#[inline]
+fn invalid_ptr<T>(addr: usize) -> *mut T {
+    #[cfg(miri)]
```

Comment:
It would be nice to have a comment explaining why we only do this on cfg(miri).

Reply:
Yeah --- it's unclear to me why this is okay. If the `#[cfg(not(miri))]` generates a Miri failure...shouldn't we not be using that code? If there's a reason that it's fine for us to essentially "cheat" at passing Miri tests, it would be nice to have that documented for future readers.

Reply:
I just followed the same example as #545. I guess it might optimize without issues here.

I note although the code generates a strict-provenance error, the original code is still permissive-provenance compatible, and code that follows permissive-provenance does not actually have UB. Miri is merely unable to check such code, and we explicitly told miri to fail if it encounters it.